pkg/async/redis/cleaner_test.go

```diff
@@ -21,35 +22,44 @@ func TestDefaultCleanCleansDeadWorkers(t *testing.T) {
 	}
 
 	// Override the default cleanActiveTaskQueue function to just count how many
-	// times it is invoked
+	// times it is invoked and then notify us when its been invoked workerCount
+	// times.
```

Comment:
This whole test was re-written ti mitigate a known flake. Why is it part of this PR? Because the flake is manifesting more frequently with the unit tests running in a container executor. I'll make it a separate PR if anyone wants to push back, but I'd rather not.